hack/dev/Dockerfile.okteto

```diff
@@ -2,12 +2,17 @@ FROM golang:1.17 as builder
 
 EXPOSE 2345
 
+ENV GO111MODULE=on
```

Comment:
this is no longer necessary since we are using Go 1.17, let's not set it?

Reply:
Should have removed this. 
I lost my calm figuring why `go build` in debug mode was downloading modules again and agin.
Later found out that our `Makefile` imports `Makefile.build` and sets the below values
https://github.com/replicatedhq/kots/blob/8985152ebb9c40cc64ff53c9e1a85f66d63af05b/Makefile.build#L1-L3

Now, In okteto debug environment, only one time modules are downloaded and later they are not downloaded again as `CGO_ENABLED=1` in `make kotsadm-debug-build` is set